src/content/docs/docs/guides/vue.mdx

```diff
@@ -0,0 +1,700 @@
+---
+title: Vue.js
+---
+
+import { Steps } from "@astrojs/starlight/components";
+import { Aside } from "@astrojs/starlight/components";
+
+In this guide, you'll learn how to create a to-do app with features to view, add, delete, and mark tasks as complete.
+
+## Prerequisites
+
+- Some experience with Vue.js
+- Node and NPM
+
+## Create New Vue Project
+
+Create new Vue project:
+
+```bash
+npm create vue@latest
+```
+
+Choose a project name. I am using `bazaar-vue-todo`.
+
+Then, answer the following prompts:
+
+```bash
+Add TypeScript? › No
+Add JSX Support? › No
+Add Vue Router for Single Page Application development? No
+Add Pinia for state management? No
+Add Vitest for Unit Testing? No
+Add an End-to-End Testing Solution? No
+Add ESLint for code quality? No
+Add Vue DevTools 7 extension for debugging? (experimental) No
+```
+
+You can choose `yes` to any prompt, though none of the options are needed for this guide.
+
+After answering all prompts, you'll see the following output:
+
+```bash title="Output"
+Scaffolding project in /<your-path>/bazaar-vue-todo...
+
+Done. Now run:
+
+  cd bazaar-vue-todo
+  npm install
+  npm run dev
+```
+
+Run those three commands, then open the new Vue app in your browser at the URL shown in the `npm run dev` output.
+
+Check out new project in browser. The default URL is `http://localhost:5173/`.
+
+## Prepare the Scaffolded Project
+
+Before you start creating your to-do app, you'll prepare the freshly scaffolded Vue project.
+
+<Steps>
+
+1. Delete the unnecessary items from the Vue starter app:
+
+   ```js
+   import { Steps } from "@astrojs/starlight/components";
+   ```
+
+   - `src/components/*`
+   - `src/assets/base.css`
+   - `src/assets/logo.svg`
+
+   You could remove the files in your code editor. Or if you prefer, in your terminal, from your project folder, run the following command:
+
+   ```bash
+   rm -rf src/components/* src/assets/base.css src/assets/logo.svg
+   ```
+
+2. Add CSS:
+
+   Open up `assets/main.css` and remove everything from the file and add the following styles:
+
+   ```css title="main.css"
+   body {
+     font-family: sans-serif;
+   }
+
+   .container {
+     margin: 0 auto;
+     max-width: 600px;
+     padding: 3em 0;
+   }
+
+   .card {
+     background-color: #eee;
+     padding: 1rem;
+   }
+
+   .todo-form {
+     margin: 2rem 0;
+   }
+
+   .todos {
+     display: flex;
+     flex-direction: column;
+     gap: 1rem;
+   }
+
+   .todo-text {
+     margin-bottom: 1rem;
+   }
+
+   .strikethrough {
+     text-decoration: line-through;
+   }
+   ```
+
+3. Prepare `App.vue`:
+
+   Next, Open up `src/App.vue` and replace its contents with the following:
+
+   ```vue title="App.vue"
+   <script setup></script>
```

Comment:
Prepare this code for the guide below by adding comments:

```
<script setup>
<!-- here we will add our TODO list logic -->
</script>
```

I just think it is nice to prepare the reader for what is to come.